### DISCUSSION TITLE
Transaction converted to Transfer not being deleted - stays duplicated

### DISCUSSION THREAD

**Discussion by PiotrEsse** [firefly-iii/firefly-iii#6984] [General]
Hi, my question can be stupid but I am inprocess of discovering Firefly3. 

I am trying to convert transaction to transfer. 
My situation: 

1. I am doing transfer between my two bank accounts in the same bank
2. my bank export transaction csv file log transfer as two atransactions 
> 2023-01-23	Incoming Transfer  	BankAccountB	500,00 PLN
> 2023-01-23	Outcoming Transfer  	BankAccountA	-500,00 PLN
3. After CSV file import (data import tool) Firefly3 properly divide both transactions - You have 500pln deposit transaction on BankAccountB, and withdrawal transactionof 500pln in BankAccountA 
4. Witchdrawal transaction entry
![image](https://user-images.githubusercontent.com/42764623/216917559-042664f2-bd87-4bfd-89f3-c36be2ef1cf2.png)
5. Deposit transaction entry
![image](https://user-images.githubusercontent.com/42764623/216917837-b0ed81f8-4a13-4fdb-ae80-64576ce47fea.png)

So far so good. 
Now I want to make rule to convert this type of  transactions to transfers as this is just transfer etween my avccounts - no incom or costs. 
1. Ive created rule: 
![image](https://user-images.githubusercontent.com/42764623/216918797-38cda17b-cb18-4498-a176-52e1bde54579.png)
2. On withdrawal transaction list this transaction disappear (I consider it ok)
![image](https://user-images.githubusercontent.com/42764623/216920197-b470970b-e1c2-4c26-b9c1-ea6ccf533a58.png)
3. On transfer transaction list there is one valid entry (there were none before)
![image](https://user-images.githubusercontent.com/42764623/216920516-d7888dea-f6a0-4d56-88c6-912929b68f3d.png)
4. but on deposit transaction list nothing have changed - I still see same transaction (also still missing source account)
![image](https://user-images.githubusercontent.com/42764623/216920825-921cd38e-3da3-415b-a240-4f6fce556903.png)

Do I created proper automatic rule? I wxpect that after converting to transfer both deposit and withdrawal transaction should dissapear - do I expect it correctly or I miss something? Please bear in mind Ive just start to discover FireFly3 and I still learn. 

Kind regards, Piotr

**Comment by JC5** [comment]
You have to make a rule to delete the second transaction as well. Firefly III will not do this for you.

**Reply by PiotrEsse** [reply]
Thank You for your quick responce. 
Can it be done in one step or I need to create second rule?
Could You please tell me how this secont rule should look like?

I am afraid if Ill make 2 rules sometime something will slip and Firefly3 wont pickup transfer rule but run transaction deletion  rule.

**Reply by JC5** [reply]
You will have to create a second rule. Set 'transaction type is' to deposit, the other factors you will have to test and set up yourself.
